<?xml version='1.0'?>
<launch>
	<arg name="model" default="$(find mantis_description)/urdf/mantis_quad.xacro"/>
	<arg name="model_name" default="mantis_uav"/>
	<arg name="frame_id" default="map"/>
	<arg name="rviz" default="true"/>
	<arg name="rqt" default="true"/>

	<rosparam command="load" ns="/mantis_uav" file="$(find mantis_description)/config/mantis_sim_params_quad.yaml"/>

	<!-- Robin Gazebo plugin specific parameters-->
	<group ns="/mantis_uav">
		<param name="prop_layout" value="quad_x4"/>
		<param name="base_arm_length" value="0.225"/>
	</group>
	<remap from="/$(arg model_name)/command/motor_pwm" to="/mavros/rc/out" />
	<remap from="/$(arg model_name)/state/pose" to="/mavros/mocap/pose"/>

	<node pkg="mantis_params" type="mantis_params_node" name="mantis_params"
	ns="/mantis_uav" clear_params="true" output="screen"/>

	<node pkg="mantis_state" type="mantis_state_node" name="mantis_state"
	ns="/mantis_uav" clear_params="true" output="screen">
		<param name="update_rate" value="100.0" />
		<param name="use_odom_angular_velocity" value="false" />

		<remap from="state/mav_state" to="/mavros/state" />
	</node>

	<!-- visualization -->
	<node pkg="mantis_kinematics" type="mantis_kinematics_forward_node" name="mantis_fk" clear_params="true" output="screen" ns="mantis_uav">
		<param name="frame_id" value="$(arg frame_id)" />
		<param name="model_id" value="$(arg model_name)" />
		<param name="update_rate" value="30.0" />

		<param name="armed_prop_velocity" value="0.6" />
		<param name="do_viz" value="false" />
		<param name="end_effector_pose" value="true" />
	</node>

	<group if="$(arg rviz)">
		<node name="rviz" pkg="rviz" args="-d $(find mantis_description)/config/display.rviz" type="rviz" respawn="false" output="screen"/>
	</group>

	<group if="$(arg rqt)">
		<node name="rqt" pkg="rqt_gui" type="rqt_gui" args="--perspective-file $(find mantis_description)/config/dashboard.perspective" respawn="false" output="screen"/>
	</group>


	<!-- Controller Setup -->
	<rosparam file="$(find mantis_gazebo)/launch/mantis_sim_joint_control.yaml" command="load"/>
	<!-- XXX: Comment out this remapping to use sim to assist with arm bench test -->
	<remap from="/mantis_uav/joint_states" to="/mantis_uav/state/joints" />

	<node name="controller_spawner" pkg="controller_manager" type="controller_manager" respawn="false"
	output="screen" ns="/mantis_uav"
	args="spawn shoulder_velocity_controller elbow_velocity_controller joint_state_controller"/>

	<!-- Gazebo -->
	<include file="$(find mantis_gazebo)/launch/mantis_sim_gazebo.launch">
		<arg name="model" value="$(arg model)" />
		<arg name="model_name" value="$(arg model_name)" />
	</include>
</launch>
